Directory.Build.props

```diff
@@ -4,6 +4,7 @@
     <ImportNetSdkFromRepoToolset>false</ImportNetSdkFromRepoToolset>
     <AssetsDir>$(MSBuildThisFileDirectory)assets\</AssetsDir>
     <VSSDKTargetPlatformRegRootSuffix>RoslynDev</VSSDKTargetPlatformRegRootSuffix>
+    <AutoGenerateAssemblyVersion>true</AutoGenerateAssemblyVersion>
```

Comment:
@tmat I am not sure how to test this locally. If I build normally, assembly versions are always 42.42.42.42 (even with `/p:Configuration=release)`. If I build with `/p:DotNetUseShippingVersions=true`, then the NuGet package version has the daily build number, i.e. `3.3.0-dev.20221.1`, but the assemblies in the package have version `3.3.0.0`.

Reply:
The repo is on a pretty old version of Arcade that likely doesn't have the feature. 
We should update to the latest Arcade and also subscribe to the `.NET Eng - Latest` channel for updates.

Reply:
@mavasani When it's done there was an open ticket about someone asking how to change the version when building locally. It might be worth answering.

Reply:
@tmat Moving to newer arcade indeed fixed this issue! Can you please review?

Reply:
It seems like CI is failing as it is unable to find NuGet packages from myget feed. It passes locally on my machine though. Any ideas on how to fix this?

Reply:
Does NuGet.Config contain all the necessary feeds?

Reply:
Feeds are specified here: https://github.com/dotnet/roslyn-analyzers/blob/d7d170a433d359941885fa22b42f3d047dd179c9/eng/Versions.props#L48-L55

Reply:
Pushed https://github.com/dotnet/roslyn-analyzers/pull/3536/commits/e881368cd23f9b3ee7f07851071de8aad155388f to move these feeds to nuget.config.